src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -285,4 +338,51 @@ public void indexAnomalyResultException(
         }
     }
 
+    private void confirmTotalRCFUpdatesFound(
+        String detectorId,
+        String taskState,
+        Long rcfTotalUpdates,
+        Long detectorIntervalInMinutes,
+        String error,
+        ActionListener<Long> listener
+    ) {
+        nodeStateManager.getAnomalyDetector(detectorId, ActionListener.wrap(detectorOptional -> {
+            if (!detectorOptional.isPresent()) {
+                listener.onFailure(new AnomalyDetectionException(detectorId, "fail to get detector"));
+                return;
+            }
+            nodeStateManager.getAnomalyDetectorJob(detectorId, ActionListener.wrap(jobOptional -> {
+                if (!jobOptional.isPresent()) {
+                    listener.onFailure(new AnomalyDetectionException(detectorId, "fail to get job"));
+                    return;
+                }
+                adTaskCacheManager.markResultIndexQueried(detectorId);
```

Comment:
Should we mark result index queried after `confirmDetectorInitStatus` done?

Reply:
yes, changed